### ISSUE TITLE
SQLAlchemy error when copying fit which has another fit projected to it

### ISSUE THREAD

**Issue by DarkFenX** [pyfa-org/Pyfa#2049]
1. Create machariel fit
2. Create another machariel fit
3. Project second fit to first
4. Copy first fit

```
/usr/lib/python3/dist-packages/sqlalchemy/orm/unitofwork.py:278: SAWarning: Object of type <Module> not in session, add operation along 'Fit._Fit__modules' will not proceed
  % (orm_util.state_class_str(state), operation, prop)
```

Might indicate that something is being wrong with how we handle modules.

**Comment by DarkFenX** [comment]
This happens when you have command or projected fit, pyfa does `eos.db.saveddata_session.flush()` as part of `forceUpdateSavedata()`. If you do `flush()` without command fit - issue reproduces, as long as you copy modules. So, this has nothing to do with projected/command fits per se.

**Comment by DarkFenX** [comment]
It indeed happened because we appended modules to a list with custom (not conforming to list interface) method, so they were not added to DB session. Fixed in 0c2c0ac6ef024ded0d0720873a07e937de3c3707
